src/pages/components/activity-graph/activity-graph.js

```diff
@@ -0,0 +1,78 @@
+import React, {useRef, useState, useEffect} from "react";
+import clsx from "clsx";
+import styles from "../../styles.module.css";
+
+import useDimensions from "react-cool-dimensions";
+import {ForceGraph3D} from "react-force-graph";
+import activityData from "./activity-data";
+
+const grey = "#FAFAFA";
+const cameraDistance = 750;
+
+/**
+ * Device Types graph
+ *
+ * @returns {object} - Types page
+ */
+function ActivityGraph() {
+  const graphRef = useRef();
+  const [counter, setCounter] = useState(0);
+  const [angle, setAngle] = useState(0);
+  const [graphData, setGraphData] = useState({nodes: [], links: []});
+  const [size, setSize] = useState({width: 100, height: 100});
+  const {observe} = useDimensions({onResize: ({width, height}) => {
+    if (size.width !== width || size.height !== height) {
+      setSize({height, width});
+    }
+  }});
+
+  // Data loading
+  useEffect(() => {
+    setGraphData(activityData);
+  }, []);
+
+  // Particle emission
+  useEffect(() => {
+    if (!graphRef.current || !graphRef.current.cameraPosition) return () => {};
+
+    const timer = setInterval(() => {
+      graphRef.current.cameraPosition({
+        x: cameraDistance * Math.sin(angle),
+        z: cameraDistance * Math.cos(angle),
+      });
+      setAngle(angle + Math.PI / 300);
+
+      setCounter(((counter + 1) % (activityData.nodes && activityData.nodes.rows.length)) || 0);
```

Comment:
I'm getting a bunch of the following in my JS console

```
activity-graph.js?051a:35 Uncaught TypeError: Cannot read properties of undefined (reading 'length')
    at eval (activity-graph.js?051a:35:1)
```

It might be worth replacing this line with (although maybe there's a bug if the counter isn't working?)

```suggestion
      setCounter(((counter + 1) % (activityData?.nodes?.rows?.length ?? 0));
```

Reply:
Yes, this is just a draft Alex asked me to put in. I'm going to put the graph behind the title and refractor the link particle generation.